okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpEngine.java

```diff
@@ -674,6 +674,7 @@ public void readResponse() throws IOException {
       return; // No network response to read.
     }
 
+    Request sentRequest = userRequest;
```

Comment:
iiiiiinteresting. I find it strange that we have a sentRequest and a networkRequest that aren't necessarily the same thing, particularly because networkRequest is supposed to be the request that is sent. But it isn't really; it's the request we sent into the interceptor chain.

I'm going to play with this change.